### DISCUSSION TITLE
Is there a way to add simple text lines between the trade item lines?

### DISCUSSION THREAD

**Discussion by s1nn3r** [stephanstapel/ZUGFeRD-csharp#301] [General]
Hey,
Thank you for the effort you put into this library! I am currently trying to achieve an X-invoice export from our ERP system. We use arbitrary texts between the line items, which do not necessarily refer to an article position; sometimes they are headings or general texts. Is there a way to write these into the positions in the X-invoice with line numbers?
When trying to add TradelineItemComments I receive this error on the document validation:
_An Invoice that contains an Invoice line (BG-25) where the Invoiced item VAT category code (BT-151) is "Not subject to VAT" shall not contain the Seller VAT identifier (BT-31), the Seller tax representative VAT identifier (BT-63) or the Buyer VAT identifier (BT-48)._
Thank you!
Marc

**Comment by stephanstapel** [comment]
There are multiple elements that can be used for different purposes. So it depends on what you want to achieve.

The validation error you mention is a famous one that has been discussed in multiple places on the internet. Unfortunately this validation doesn't make sense (in this context).

**Reply by Tissid04** [reply]
Hey @s1nn3r,

How did you handle this?

@stephanstapel, you mentioned there are multiple elements that can be used. Do you have a proposal for just a text line (not depending on a line before or after)?

Thanks for helping out!

**Reply by stephanstapel** [reply]
You can use AddTradeLineCommentItem().
You can also use the line status as explained on the main documentation page using
https://github.com/stephanstapel/ZUGFeRD-csharp/blob/fa17b6425470ade812802cb95fab1209feab7830/ZUGFeRD/LineStatusCodes.cs#L698

**Reply by Tissid04** [reply]
When I use the AddTradeLineCommentItem function, I receive the well-known validation error: 

[BR-O-02] - An Invoice that contains an Invoice line (BG-25) where the Invoiced item VAT category code (BT-151) is "Not subject to VAT" shall not contain the Seller VAT identifier (BT-31), the Seller tax representative VAT identifier (BT-63), or the Buyer VAT identifier (BT-48). 

It works through Line Status, thank you very much.

**Reply by stephanstapel** [reply]
@Tissid04 : Yes, that makes sense.
Can you please change this line:
https://github.com/stephanstapel/ZUGFeRD-csharp/blob/ade3d20b98324c3e9b2f151a1d624aedfb5f36d2/ZUGFeRD/InvoiceDescriptor.cs#L1140

from O to E and validate again? That would help other users a lot, I guess!

**Reply by Tissid04** [reply]
Hello Stephan,

I have implemented the change, but without success. I am now getting the following error message: 

[BR-E-01] - An Invoice that contains an Invoice line (BG-25), a Document level allowance (BG-20), or a Document level charge (BG-21) where the VAT category code (BT-151, BT-95, or BT-102) is "Exempt from VAT" shall contain exactly one VAT breakdown (BG-23) with the VAT category code (BT-118) equal to "Exempt from VAT".

**Reply by stephanstapel** [reply]
Thanks for testing it! I will dive a bit deeper into this topic as well.

**Comment by Ole-DD** [comment]
Hi,
I tried to create comment lines, too:
```
var result = desc.AddTradeLineCommentItem("1", "Das ist ein Kommentar!", string.Empty, string.Empty);
result.SetLineStatus(LineStatusCodes.ForInformation, LineStatusReasonCodes.INFORMATION);

```
Without SetLineStatus I got the same results. 
I tried SetLineStatus with XRechnung and comfort profile.
Unfortunately the option "LineStatusCode" is only avaliable in extended profile.

Any other chance to create comment lines?
Olaf
